pennylane_lightning/core/src/measurements/MeasurementsBase.hpp

```diff
@@ -76,10 +76,15 @@ template <class StateVectorT, class Derived> class MeasurementsBase {
 
     /**
      * @brief Randomly set the seed of the internal random generator
+     * If catalyst already provides a random generator, use the provided one
      *
-     * @param seed Seed
+     * @param catalyst_rng Seed
      */
-    void setRandomSeed() {
+    void setRandomSeed(std::mt19937 *catalyst_rng = nullptr) {
+        if (catalyst_rng){
+            this->rng = *catalyst_rng;
+            return;
```

Comment:
Do we need `return` here?

Reply:
I think it is, we don't want to change the `mt19937` object once it's set (the `setSeed` below will change it)

Reply:
No longer relevant now that the two functionality is split into two functions